packages/salesforce-adapter/src/transformers/reference_mapping.ts

```diff
@@ -106,6 +106,15 @@ const ReferenceSerializationStrategyLookup: Record<
     serialize: async ({ ref }) => ref.elemID.name,
     lookup: val => val,
   },
+  customLabel: {
+    serialize: async ({ ref }) => `$Label.${ref.elemID.name}`,
```

Comment:
1. if you're using `API_NAME_SEPARATOR` below you should use it here as well to make sure this is symmetric (not sure if it should be used here tbh, since this is not the usual api-name syntax IIUC so it "just happens" to also be the same character)
2. we shouldn't serialize based on `elemID.name` - `mapKey`, but you'll see none of the other strategies are using it. 
there's a nuance here - the elem id may not be what you expect - both because it can be edited by the user, and because once an instance gets its elem id the elem id doesn't change even if it would have received another one on a "first fetch" (might be easier to discuss this one IRL or I can send you some pointers if you want).
so this should be
```suggestion
    serialize: async ({ ref }) => `$Label${API_NAME_SEPARATOR}${await safeApiName({ ref, path }}`,
```
(and the function should be async) like in the other examples